src/app/services/dataset/dataset_bucket_service.py

```diff
@@ -18,18 +18,23 @@ def get_and_validate_dataset(self, filename: str) -> RawDataset:
         Parameters:
         filename: name of file being retrieved from bucket
         """
-        DatasetValidatorService.validate_file_is_json(filename)
+        isValid, message = DatasetValidatorService.validate_file_is_json(filename)
+        if not isValid:
+            if config.AUTODELETE_DATASET_BUCKET_FILE is True:
+                self.try_delete_bucket_file(filename)
 
-        raw_dataset = self.dataset_bucket_repository.get_dataset_file_as_json(filename)
+            raise RuntimeError(message)
```

Comment:
better to throw a more accurate error rather than run time like TypeError rather than a Runtime. i guess this wont use the custom error handler since im unsure how that would get handled with the cloud function. 
https://docs.python.org/3/library/exceptions.html#TypeError

Reply:
The monitoring module is only capturing the RuntimeError from cloud function, so that's why I have not choose to use a more specific error type.